plugins/modules/system/alternatives.py

```diff
@@ -90,10 +135,210 @@
+                link=self.current_link,
+                subcommands=self.current_alternatives[self.path].get('subcommands')
+            ))
+
+        if self.mode_present:
+            # Check if we need to (re)install
+            if (
+                self.path not in self.current_alternatives or
+                self.current_alternatives[self.path].get('priority') != self.priority or
+                not all(s in self.subcommands for s in self.current_alternatives[self.path].get('subcommands')) or
+                not all(s in self.current_alternatives[self.path].get('subcommands') for s in self.subcommands)
+            ):
+                self.install()
+
+            # Check if we need to set the preference
+            if self.mode_selected and self.current_path != self.path:
+                self.set()
+
+            # Check if we need to reset to auto
+            if self.mode_auto and self.current_mode == 'manual':
+                self.auto()
+        else:
+            # Check if we need to uninstall
+            if self.path in self.current_alternatives:
+                self.remove()
+
+        self.module.exit_json(**self.result)
+
+    def install(self):
+        if not os.path.exists(self.path):
+            self.module.fail_json(msg="Specified path %s does not exist" % self.path)
+        if not self.link:
+            self.module.fail_json(msg='Needed to install the alternative, but unable to do so as we are missing the link')
+
+        cmd = [self.UPDATE_ALTERNATIVES, '--install', self.link, self.name, self.path, str(self.priority)]
+
+        if self.subcommands:
+            subcommands = [['--slave', subcmd['link'], subcmd['name'], subcmd['path']] for subcmd in self.subcommands]
+            cmd += [item for sublist in subcommands for item in sublist]
+
+        self.result['changed'] = True
+        self.result['msg'] = "Install alternative '%s' for '%s'" % (self.path, self.name)
+
+        if not self.module.check_mode:
+            self.module.run_command(cmd, check_rc=True)
+
+        if self.module._diff:
+            self.result['diff']['after'] = dict(
+                path=self.path,
+                priority=self.priority,
+                link=self.link,
+                subcommands=self.subcommands
+            )
+
+    def remove(self):
+        cmd = [self.UPDATE_ALTERNATIVES, '--remove', self.name, self.path]
+        self.result['changed'] = True
+        self.result['msg'] = "Remove alternative '%s' from '%s'" % (self.path, self.name)
+
+        if not self.module.check_mode:
+            self.module.run_command(cmd, check_rc=True)
+
+        if self.module._diff:
+            self.result['diff']['after'] = dict(state=AlternativeState.ABSENT)
+
+    def set(self):
+        cmd = [self.UPDATE_ALTERNATIVES, '--set', self.name, self.path]
+        self.result['changed'] = True
+        self.result['msg'] = "Set alternative '%s' for '%s'" % (self.path, self.name)
```

Comment:
Right now you're always overwriting `self.result['msg']`. How about storing these messages in a list, and setting `self.result['msg'] = ' '.join(self.messages)` at the end of `run()`? (Also the individual messages should better end with a period in this case.)

Reply:
updated